daemon/logger/logger.go

```diff
@@ -71,7 +73,9 @@ func NewLogWatcher() *LogWatcher {
 
 // Close notifies the underlying log reader to stop.
 func (w *LogWatcher) Close() {
-	close(w.closeNotifier)
+	w.closeOnce.Do(func() {
```

Comment:
Told about this https://github.com/docker/docker/blob/master/daemon/monitor.go#L262
Once ok too.